app/src/test/java/com/weatherxm/util/NumberUtilsTest.kt

```diff
@@ -43,19 +58,74 @@ class NumberUtilsTest : BehaviorSpec({
+                        NumberUtils.formatNumber(1000.4, 1) shouldBe "1,000.4"
+                    }
                 }
                 then("roundToDecimals should return a valid number") {
                     NumberUtils.roundToDecimals(100.123456) shouldBe 100.1F
                     NumberUtils.roundToDecimals(100.15) shouldBe 100.2F
                     NumberUtils.roundToDecimals(100.02) shouldBe 100.0F
                 }
-                then("roundToInt should return a valid number") {
-                    NumberUtils.roundToInt(1.0) shouldBe 1
-                    NumberUtils.roundToInt(1.49) shouldBe 1
-                    NumberUtils.roundToInt(1.5) shouldBe 2
-                    NumberUtils.roundToInt(1.51) shouldBe 2
+            }
+        }
+    }
+
+    context("Formatting tokens as a text with min 2 and max 3 decimals") {
+        given("a token amount") {
+            When("it is null") {
+                then("the formatter should return an empty string") {
+                    formatTokens(null) shouldBe EMPTY_VALUE
+                }
+            }
+            When("it is zero") {
+                then("the formatter should return 0.00") {
+                    formatTokens(0F) shouldBe "0.00"
+                }
+            }
+            When("it is an integer") {
+                and("it is == 1000") {
+                    then("return the integer with thousands separator and 2 decimals") {
+                        formatTokens(1000F) shouldBe "1,000.00"
+                    }
+                }
+                and("it is < 1000") {
+                    then("the formatter should return the integer with 2 decimals e.g. 10.00") {
+                        formatTokens(10F) shouldBe "10.00"
+                    }
+                }
+            }
+            When("it has up to 2 decimals") {
+                then("the formatter should return the amount with 2 decimals") {
+                    formatTokens(10.1F) shouldBe "10.10"
+                    formatTokens(10.01F) shouldBe "10.01"
+                }
+            }
+            When("it has >=3 decimals") {
+                then("the formatter should return the amount with 2 decimals") {
+                    formatTokens(10.001F) shouldBe "10.00"
+                    formatTokens(10.005F) shouldBe "10.01"
+                }
+            }
+        }
+    }
+
+    context("Convert WEI -> ETH correctly and show the correct amount") {
+        given("a WEI amount") {
+            When("it is zero") {
+                then("the convertor should return zero") {
+                    weiToETH(BigDecimal.ZERO) shouldBe BigDecimal.ZERO
+                }
+            }
+            When("it is bigger than zero") {
+                then("the validator should return make the conversion and format the tokens") {
+                    formatTokens(weiToETH(BigDecimal.valueOf(10000000000000000))) shouldBe "0.01"
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The test case for WEI to ETH conversion only tests a single non-zero scenario, which may not be sufficient to ensure the function's correctness across different input ranges.
</div>
<div id="fix">
<b>Fix:</b> Add more test cases to cover a wider range of input values for the WEI to ETH conversion, including edge cases and boundary values.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -126,6 +126,18 @@ class NumberUtilsTest : BehaviorSpec({
          When("it is bigger than zero") {
              then("the validator should return make the conversion and format the tokens") {
                  formatTokens(weiToETH(BigDecimal.valueOf(10000000000000000))) shouldBe "0.01"
 +                // Test with 1 WEI (smallest unit)
 +                formatTokens(weiToETH(BigDecimal.ONE)) shouldBe "0.00"
 +                // Test with 1 ETH in WEI
 +                formatTokens(weiToETH(BigDecimal.valueOf(1000000000000000000))) shouldBe "1.00"
 +                // Test with a large WEI amount
 +                formatTokens(weiToETH(BigDecimal.valueOf(1234567890000000000000))) shouldBe "1,234.57"
 +                // Test with a very small non-zero WEI amount
 +                formatTokens(weiToETH(BigDecimal.valueOf(1000))) shouldBe "0.00"
 +            }
 +            and("the input is the maximum possible WEI value") {
 +                then("the conversion should not overflow") {
 +                    weiToETH(BigDecimal("115792089237316195423570985008687907853269984665640564039457584007913129639935")) shouldBe BigDecimal("115792089237316195423570985008687907853269984665640564039457.584007913129639935")
              }
          }
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged